src/lib.rs

```diff
@@ -88,6 +89,33 @@ macro_rules! try_get_param {
 }
 
 
+/// Helper function for returning a result to the Relying Party.
```

Comment:
In this commit, I'd prefer to have `return_to_relier()` change in place (that is, don't change the order here), so it's easier to see what's going on here. You can of course move it around after that.